api/opencl_platform_layer.asciidoc

```diff
@@ -2178,17 +2178,17 @@ include::{generated}/api/version-notes/CL_CONTEXT_PROPERTIES.asciidoc[]
         {clCreateContextFromType}.
 
         If the _properties_ argument specified in {clCreateContext} or
-        {clCreateContextFromType} used to create _context_ is not `NULL`, the
-        implementation must return the values specified in the properties
-        argument.
+        {clCreateContextFromType} used to create _context_ was not `NULL`,
+        the implementation must return the values specified in the
+        properties argument in the same order and without including
+        additional properties.
 
         If the _properties_ argument specified in {clCreateContext} or
-        {clCreateContextFromType} used to create _context_ is `NULL`, the
-        implementation may return either a _param_value_size_ret_ of 0
-        i.e. there is no context property value to be returned or can return
-        a context property value of 0 (where 0 is used to terminate the
-        context properties list) in the memory that _param_value_ points
-        to.
+        {clCreateContextFromType} used to create _context_ was `NULL`, the
+        implementation may return either a
+        _param_value_size_ret_ of 0 (i.e. there are no properties to be
+        returned), or the implementation may return a property value of 0
+        (where 0 is used to terminate the properties list).
```

Comment:
I think this is the last open question referenced in the description.

It seems slightly complicated to to have two possible behaviours in this case.  I don't really have a preference between them, if forced to a side I'd go with the `0` size to match the `NULL` input.

Reply:
Discussed in the August 11th teleconference and agreed to return a size of zero for a `NULL` input.  I will update this PR to reflect this.

Reply:
Fixed!